sdk/core/core-rest-pipeline/src/pipeline.ts

```diff
@@ -119,9 +119,12 @@ class HttpPipeline implements Pipeline {
   private _policies: PipelineDescriptor[] = [];
   private _orderedPolicies?: PipelinePolicy[];
 
+  private _cache: WeakMap<HttpClient, SendRequest>;
+
   private constructor(policies: PipelineDescriptor[] = []) {
     this._policies = policies;
     this._orderedPolicies = undefined;
+    this._cache = new WeakMap();
```

Comment:
perhaps we should add a unit test to call in sequence with different httpClients just to ensure the caching never gets confused?

Reply:
Yeah, that sounds good. I was thinking a few minutes ago that I hadn't tested that per-client caching works correctly.